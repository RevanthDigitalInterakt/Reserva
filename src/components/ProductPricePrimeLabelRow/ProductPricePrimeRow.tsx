import React, { useMemo } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { integerPart, decimalPart } from '../../utils/numberUtils';
import { styles } from './ProductPricePrimeRow.styles';

interface IProps {
  installments?: { value: number; number: number; };
  currency?: string;
  discountTag?: number;
  priceWithDiscount?: number;
  price: number;
}

const defaultCurrency = 'R$';

const ProductPricePrimeRow = ({
  installments,
  currency = defaultCurrency,
  discountTag,
  priceWithDiscount,
  price,
}: IProps) => {
  const finalPrice = useMemo(() => (
    discountTag && priceWithDiscount ? priceWithDiscount : price
  ), [discountTag, priceWithDiscount, price]);

  const container = StyleSheet.flatten([
    styles.wrapper,
    installments ? styles.start : styles.between,
  ]);

  const flag = StyleSheet.flatten([
    styles.containerPrime,
    installments && styles.ml,
  ]);

  return (
    <View style={container}>
      <View style={{ flexDirection: 'row', alignItems: 'center' }}>
        <Text style={styles.textPrice}>
          {`${currency} `}
        </Text>

        <Text style={styles.textPrice}>
          {`${integerPart(finalPrice)},`}
        </Text>

        <Text style={styles.textPrice}>
          {`${decimalPart(finalPrice)}`}
        </Text>

        {installments && (
          <>
            <View style={styles.divider} />
            <Text style={[styles.textInstallments, { fontSize: 12 }]}>
              {installments.number}
              x
            </Text>

            <Text style={styles.textInstallments}>
              {` ${currency} ${integerPart(installments.value)},`}
            </Text>

            <Text style={styles.textInstallments}>
              {`${decimalPart(installments.value)}`}
            </Text>
          </>
        )}

        <View style={[flag, { marginLeft: 5 }]}>
          <Text style={styles.labelPrime}>Prime</Text>
        </View>
      </View>
    </View>
  );
};

export default ProductPricePrimeRow;
