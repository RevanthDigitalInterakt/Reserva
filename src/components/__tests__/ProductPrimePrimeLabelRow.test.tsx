import React from 'react';
import {
  render,
} from '@testing-library/react-native';
import ProductPricePrimeRow from '../ProductPricePrimeLabelRow/ProductPricePrimeRow';

describe('ProductPrimePrimeLabelRow', () => {
  it('should match with the snapshot', async () => {
    const Component = (
      <ProductPricePrimeRow
        installmentsNumber={6}
        installmentsPrice={200}
        discountTag={100}
        priceWithDiscount={150}
        price={1000}
      />
    );

    const { toJSON } = render(Component);

    expect(toJSON()).toMatchSnapshot();
  });
});
