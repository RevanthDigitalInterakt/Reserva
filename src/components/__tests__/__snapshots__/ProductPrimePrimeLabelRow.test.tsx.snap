// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPrimePrimeLabelRow should match with the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#A01E21",
          "fontFamily": "ReservaSans-Medium",
          "fontSize": 12,
          "letterSpacing": 0.05,
        }
      }
    >
      R$ 
    </Text>
    <Text
      style={
        Object {
          "color": "#A01E21",
          "fontFamily": "ReservaSans-Medium",
          "fontSize": 12,
          "letterSpacing": 0.05,
        }
      }
    >
      150,
    </Text>
    <Text
      style={
        Object {
          "color": "#A01E21",
          "fontFamily": "ReservaSans-Medium",
          "fontSize": 12,
          "letterSpacing": 0.05,
        }
      }
    >
      00
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#A01E21",
            "padding": 2,
          },
          Object {
            "marginLeft": 5,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "ReservaDisplay-Regular",
            "fontSize": 8,
          }
        }
      >
        Prime
      </Text>
    </View>
  </View>
</View>
`;
